dkg/dkg.go

```diff
@@ -181,8 +192,60 @@ func setupP2P(ctx context.Context, datadir string, p2pConf p2p.Config, peers []p
 	}, nil
 }
 
-func aggSignLockHash(_ context.Context, _ kcTransport, _ cluster.Lock) ([]byte, error) {
-	// TODO(corver): Implement lock hash signing by all DVs and aggregation.
+// aggSignLockHash returns the aggregated multi signature of the lock hash
+// signed by all the distributed validator group private keys.
+func aggSignLockHash(ctx context.Context, tcpNode host.Host, peerIdx int,
+	peers []peer.ID, lock cluster.Lock, shares []share,
+) (*bls_sig.MultiSignature, error) {
+	signedSet, err := signLockHash(lock, shares)
+	if err != nil {
+		return nil, err
+	}
+
+	sigChan := make(chan *bls_sig.Signature, len(lock.Validators))
+	db := parsigdb.NewMemDB(lock.Threshold)
+	exchange := parsigex.NewParSigEx(tcpNode, peerIdx, peers)
+	db.SubscribeInternal(exchange.Broadcast)
+	db.SubscribeThreshold(makeSigAgg(sigChan))
+	exchange.Subscribe(db.StoreExternal)
```

Comment:
using core workflow components for all the heavy lifting.

Note that the generic way these components have been implemented, allows them to be used in a different use-case here in DKG.

Reply:
that is the select function below

Reply:
I don't understand, which line?

Reply:
these methods only wires stuff, the output is below at:
```
		    case sig := <-sigChan:
			sigs = append(sigs, sig)
		}
		if len(sigs) == len(lock.Validators) {
			break
		}
```

Reply:
yes, my question if it the stuff is unwired or if they keep wired, is because this func is only called once?

Reply:
those components are pure stateless functions, so there is nothing to shutdown.

Reply:
There are no goroutines inside those components.

Reply:
is this like a mini-DI container?

Reply:
I am seeing that there is this events subscriptions, and I don't see the unsubscription, so I guess it is not too weird too ask. If this function is going to be run in the background in a goroutine as long charon is running I guess there is no need for unsubscription. Is that right?

Reply:
@dB2510 so you know the answer to this?

Reply:
yeah that's right, there's no need to unsubscribe. So basically in charon we follow functional programming paradigm where we keep our functions/methods to be stateless and only introduce statefulness whenever necessary. We don't need unsubscription because we aren't persisting anything. So when that function goes out of scope, we are done. Go's GC will take care of that.
On the other hand, we introduce statefulness for functions mainly by returning functions i.e, closures.

Am I right, @corverroos ?

Reply:
exactly right! 🚀

Reply:
> Is this intended for some reason? or just a temporary shortcut that will be refactored in the future?
yes, this is temporary. You can refer here for next steps: #522

Reply:
ok 👍